known-bugs.txt

```diff
@@ -29,436 +29,429 @@ reasons why we think that fixing them is infeasible. We might make some
+        anything about this. Please report these crashes to the authors
+        of OpenMSX so the crash causing songs can be removed or fixed.
+
+Crashes when run in a VM using Parallels Desktop [#4003]:
+        When the Windows version of OpenTTD is executed in a VM under
+        Parallels Desktop a privileged instruction exception may be thrown.
+        As OpenTTD works natively on macOS as well as natively on Windows and
+        these native builds both don't exhibit this behaviour this crash is
+        most likely due to a bug in the virtual machine, something out of
+        the scope of OpenTTD. Most likely this is due to Parallels Desktop
+        lacking support for RDTSC calls. The problem can be avoided by using
+        other VM-software, Wine, or running natively on macOS.
+
+OpenTTD hangs when started on 32-bit Windows [#4083]:
+        Under some circumstances OpenTTD might hang for hours on the
+        initialisation of the music driver. The exact circumstances are
+        unknown except that it is the "dmusic" music driver that has the
+        problem and that the "Win32" music driver does not. As a result
+        using the "Win32" music driver will work around this issue.
+
+        As the exact circumstances are unknown, and the obvious
+        configuration settings related to the music driver are at their
+        default we are not able to detect this failure, except when Windows'
+        music initialisation function returns after several hours and then
+        there is no point in switching the music driver anymore.
+        The reason we still use the "Win32" music driver as default are
+        described in the "Long delay between switching music/song" section
+        of this document.
+
+Entry- and exit signals are not dragged [#4378]:
+        Unlike all other signal types, the entry- and exit signals are not
+        dragged but instead normal signals are placed on subsequent track
+        sections. This is done on purpose as this is the usually more
+        convenient solution. There are little to no occasions where more
+        than one entry or exit signal in a row are useful. This is different
+        for all other signal types where several in a row can serve one
+        purpose or another.
+
+Station build date is incorrect [#4415]:
+        The tile query tool will show the date of the last (re)construction
+        at the station and not the date of the first construction. This is
+        due to compatability reasons with NewGRFs and the fact that it is
+        wrong to say that the station is built in a particular year when it
+        was completely destroyed/rebuilt later on.
+        The tile query tool can be fixed by changing the "Build date" text
+        to "Date at which the last (re)construction took place" but this is
+        deemed too specific and long for that window.
+
+Can't change volume inside OpenTTD [#4416]:
```

Comment:
Partially fixed, both Win32 music drivers and the new Fluidsynth driver (for Linux) now support music volume control, and independent of sound effects volume.

Reply:
Music drivers that do not support volume control:
- BeOS
- extmidi
Music drivers where volume control might affect both sound effects and music:
- Allegro
Music drivers with fully functioning, independent volume control:
- Win32
- DMusic
- Fluidsynth (new default on Linux)
- Cocoa (macOS)
- QuickTime
- OS/2
- libtimidity (PSP)